automation/taskcluster/lib/tasks.py

```diff
@@ -127,7 +172,78 @@ def craft_ui_tests_task(self):
+                bucket_name = 'maven-snapshot-staging'
+                bucket_public_url = 'https://maven-snapshots.stage.mozaws.net/'
+            else:
+                bucket_name = 'maven-snapshot-production'
+                bucket_public_url = 'https://snapshots.maven.mozilla.org/'
+        else:
+            if is_staging:
+                bucket_name = 'maven-staging'
+                bucket_public_url = 'https://maven-default.stage.mozaws.net/'
+            else:
+                bucket_name = 'maven-production'
+                bucket_public_url = 'https://maven.mozilla.org/'
+
+        payload = {
+            "artifactMap": [{
+                "locale": "en-US",
+                "paths": {
+                    artifact['taskcluster_path']: {
+                        'checksums_path': '',  # TODO beetmover marks this as required even though it's not needed here
+                        'destinations': [artifact['maven_destination']]
+                    } for artifact in (build_artifacts)
+                },
+                "taskId": build_task_id,
+            }, {
+                "locale": "en-US",
+                "paths": {
+                    artifact['taskcluster_path']: {
+                        'checksums_path': '',  # TODO beetmover marks this as required even though it's not needed here
+                        'destinations': [artifact['maven_destination']]
+                    } for artifact in (sign_artifacts)
+                },
+                "taskId": sign_task_id,
+            }],
+            "maxRunTime": 600,
+            "upstreamArtifacts": [{
+                'paths': [artifact['taskcluster_path'] for artifact in build_artifacts],
+                'taskId': build_task_id,
+                'taskType': 'build',
+            }, {
+                'paths': [artifact['taskcluster_path'] for artifact in sign_artifacts],
+                'taskId': sign_task_id,
+                'taskType': 'signing',
+            }],
+            "releaseProperties": {
+                "appName": "snapshot_components" if is_snapshot else "components",
+            },
+        }
+
+        return self._craft_default_task_definition(
+            self.beetmover_worker_type,
+            'scriptworker-prov-v1',
+            dependencies=[build_task_id, sign_task_id, wait_on_all_sign_tasks_id],
+            routes=[],
+            scopes=[
+                "project:mobile:android-components:releng:beetmover:bucket:{}".format(bucket_name),
+                "project:mobile:android-components:releng:beetmover:action:push-to-maven",
+            ],
+            name="Android Components - Publish Module :{} via beetmover".format(component_name),
+            description="Publish release module {} to {}".format(component_name, bucket_public_url),
+            payload=payload
+        )
+
+    # XXX: temporarily until we add signing support in snapshots in bug 155879
+    # in early Q3
+    def craft_snapshot_beetmover_task(
```

Comment:
I forked and kept the old implementation for as long as we have snapshots still un-signed. Once that support is added, I'll attempt to meld the two in the same function as it currently is. At this point they were far too different to mess around the functions so I preffered this way. It's temporarily anyway.